docs/_posts/examples/3400-01-24-drag-a-point.html

```diff
@@ -0,0 +1,166 @@
+---
+layout: example
+category: example
+title: "Create a draggable point"
+description: "Drag the point to a new location on a map and populates its coordinates in a display."
+---
+
+<style>
+.coordinates {
+    background: rgba(0,0,0,0.5);
+    color: #fff;
+    position: absolute;
+    bottom: 10px;
+    left: 10px;
+    padding:5px 10px;
+    margin: 0;
+    font-size: 11px;
+    line-height: 18px;
+    border-radius: 3px;
+    display: none;
+}
+</style>
+
+<div id='map'></div>
+<pre id='coordinates' class='coordinates'></pre>
+
+<script>
+// Holds mousedown state for events. if this
+// flag is active, we move the point on `mousemove`.
+var onDrag;
+
+// Is the cursor over a point? if this
+// flag is active, we listen for a mousedown event.
+var overPoint;
+
+var coordinates = document.getElementById('coordinates');
+var map = new mapboxgl.Map({
+    container: 'map',
+    style: 'mapbox://styles/mapbox/streets-v8',
+    center: [0, 0],
+    zoom: 2
+});
+
+var canvas = map.getCanvasContainer();
+
+// Print the coordinates of where the point had
+// finished being dragged to on the map.
+function onDragEnd(coords) {
+    coordinates.style.display = 'block';
+    coordinates.innerHTML = 'Longitude: ' + coords.lng + '<br />Latitude: ' + coords.lat;
+}
+
+var geojson = {
+    "type": "FeatureCollection",
+    "features": [{
+        "type": "Feature",
+        "geometry": {
+            "type": "Point",
+            "coordinates": [0, 0]
+        }
+    }]
+};
+
+function getCoordinates(e) {
```

Comment:
What would it take to be able to implement dragging using map mouse events, rather than raw DOM events? The benefit of map events is that they come with coordinates pre-calculated, so `getCoordinates` wouldn't be necessary.